client/lib/plugins/sanitize-section-content.js

```diff
@@ -0,0 +1,190 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import validUrl from 'valid-url';
+
+const isAllowedTag = tagName => {
+	switch ( tagName ) {
+		case '#text':
+		case 'a':
+		case 'b':
+		case 'blockquote':
+		case 'code':
+		case 'div':
+		case 'em':
+		case 'h1':
+		case 'h2':
+		case 'h3':
+		case 'h4':
+		case 'h5':
+		case 'h6':
+		case 'i':
+		case 'img':
+		case 'li':
+		case 'ol':
+		case 'p':
+		case 'span':
+		case 'strong':
+		case 'ul':
+			return true;
+		default:
+			return false;
+	}
+};
+
+const isAllowedAttr = ( tagName, attrName ) => {
+	if ( 'a' === tagName && 'href' === attrName ) {
+		return true;
+	}
+
+	if ( 'iframe' === tagName ) {
+		switch ( attrName ) {
+			case 'class':
+			case 'type':
+			case 'height':
+			case 'width':
+			case 'src':
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	if ( 'img' === tagName && ( 'alt' === attrName || 'src' === attrName ) ) {
+		return true;
+	}
+
+	return false;
+};
+
+const isValidYoutubeEmbed = node => {
+	if ( node.nodeName.toLowerCase() !== 'iframe' ) {
+		return false;
+	}
+
+	if ( node.getAttribute( 'class' ) !== 'youtube-player' ) {
+		return false;
+	}
+
+	if ( node.getAttribute( 'type' ) !== 'text/html' ) {
+		return false;
+	}
+
+	const link = document.createElement( 'a' );
```

Comment:
I wonder how much overhead is in `document.createElement` versus something like a regex check.

Reply:
way more I'm guessing, but it's incredibly difficult to do this right without a full-blown parser. there's one in the browser and we can pull in other libraries.

RegExps will fail us here _and_ be incredibly confusing to debug.

Reply:
It's sad we can't use a simple `new URL(...).hostname`, because IE doesn't have support for `URL` :(

There's a `url-polyfill` package that's also implemented using an `a` element with `href`. It does some [extra tricks](https://github.com/lifaon74/url-polyfill/blob/master/url-polyfill.js#L183-L186) with the element, like actually appending it to the document or forcing refresh with `el.href = e.href`. I'm wondering why they need to do this? To force some lazy parsing of the `href` attribute?

@samouri reported an issue where a WooCommerce YouTube video was not displayed -- maybe it's related? Was it removed because the `href` attribute failed to validate?

Reply:
> @samouri reported an issue where a WooCommerce YouTube video was not displayed -- maybe it's related? Was it removed because the href attribute failed to validate?

I can't repro the issue anymore.

Reply:
>  It does some extra tricks with the element, like actually appending it to the document or forcing refresh with el.href = e.href. I'm wondering why they need to do this?

Possibly to catch some `<base href="…" />` cases.

Reply:
> It's sad we can't use a simple new URL(...).hostname

There's also a polyfill for [node URL.parse](https://nodejs.org/docs/latest-v8.x/api/url.html) available though I'm not sure it'd be faster.

Reply:
Let’s remember that almost _any_ additional import will end up as more code than his little bit of duplication. URL support would be great, but in the meantime do we have a real problem with using the browser facilities as is?

Reply:
>  but in the meantime do we have a real problem with using the browser facilities as is

Shouldn't be an issue for most cases. Just noting the option.